cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/parser/ArgumentParser.java

```diff
@@ -301,16 +322,31 @@ ArgumentNode parseInternal(ParseContext ctx) throws ParseException {
                     }
                     prev = 0;
                 } else if (cur == ' ' || cur == '\t' || cur == '"' || cur == '(' || cur == ')' || cur == '#')  {
-                    return new ArgumentNode(sb.toString(), expressions, start, end);
-                } else if (cur == '$') {
+                    if (sb.length() != 0) {
+                        expressions.add(new ConstantNode(sb.toString(), start, end));
+                    }
+
+                    return new ArgumentNode(expressions);
+                } else if (cur == '$') {// todo add canParse check?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)